### PR TITLE
Ab#81435 find seafarer

### PR DESCRIPTION
**Pull request by tc-chris**

### CODE REVIEW

**Comment by tc-tibo** [comment]
Merging into `master`?

File: src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        public void Search()
+        {
+
+            State.SearchCriteria = SearchCriteria;
+
+            State.ApplicantSearchResult = MpdisService.Search(SearchCriteria);
+
+            NavigationManager.NavigateTo("/SearchResults");
+
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs:57]:
```suggestion
```

File: src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }
+
+            State.ApplicantSearchResult = null;
+            EditContext = new EditContext(SearchCriteria);
+
+        }
+
+        public void Search()
+        {
+
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs:51]:
```suggestion
```

File: src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CSF.SRDashboard.Client.Pages
+{
+    public partial class SearchResults
+    {
+        [Inject]
+        public SessionState State { get; set; }
+        [Inject]
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs:20]:
```suggestion

        [Inject]
```

File: src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            if (tableSettings != null)
+            {
+                TableRef.ResetTableSettings(tableSettings);
+                tableSettings = null;
+            }
+        }
+        public void OnFilterChanged(TableSettings<ApplicantSearchResultItem> settings)
+        {
+
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs:61]:
```suggestion
```

File: src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            if (TableRef != null)
+            {
+                memoryCache.Set("TableSettings", settings);
+            }
+        }
+        protected void HandleHeaderFilterChanged()
+        {
+            StateHasChanged();
+        }
+
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Pages/SearchResults.razor.cs:76]:
```suggestion
```

File: src/CSF.SRDashboard.Client/Services/ResultsService.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+using System.Threading.Tasks;
+
+namespace CSF.SRDashboard.Client.Services
+{
+    public class ResultsService : IResultsService
+    {
+        public List<ApplicantSearchResultItem> list { get; set; }
+        public ResultsService()
+        {
+        }
```
Comment (AlexandreSanscartier) [inline src/CSF.SRDashboard.Client/Services/ResultsService.cs:14]: Remove empty constructor.
```suggestion
```

File: src/MPDIS.API.Wrapper/Services/MPDIS/Entities/ApplicantSearchResult.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MPDIS.API.Wrapper.Services.MPDIS.Entities
+{
+    public class ApplicantSearchResult
+    {
+
```
Comment (AlexandreSanscartier) [inline src/MPDIS.API.Wrapper/Services/MPDIS/Entities/ApplicantSearchResult.cs:11]:
```suggestion
```

File: src/MPDIS.API.Wrapper/Services/MPDIS/Entities/ApplicantSearchResult.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        [JsonProperty("totalCount")]
+        public long TotalCount { get; set; } = 0;
+
+
```
Comment (AlexandreSanscartier) [inline src/MPDIS.API.Wrapper/Services/MPDIS/Entities/ApplicantSearchResult.cs:15]:
```suggestion
```

**Review by AlexandreSanscartier — CHANGES_REQUESTED** [review]
Looks good mostly changes to spacing. Good work.

File: src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            base.OnInitialized();
+
+            if (State.SearchCriteria != null)
+            {
+                SearchCriteria = State.SearchCriteria;
+            }
+
+            State.ApplicantSearchResult = null;
+            EditContext = new EditContext(SearchCriteria);
+
```
[the remark this answers is not in the document]
Comment (tc-chris) [reply src/CSF.SRDashboard.Client/Pages/FindSeafarer.razor.cs]: Fixed

**Review by AlexandreSanscartier — APPROVED** [review]

**Pull request merged**